go/tendermint/apps/staking/staking.go

```diff
@@ -323,15 +459,19 @@ func (app *stakingApplication) queryGenesis(s, r interface{}) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	ledger := make(map[signature.MapKey]*staking.GenesisLedgerEntry)
+	ledger := make(map[signature.MapKey]*staking.Account)
 	for _, acctID := range accounts {
 		acct := state.account(acctID)
-		ledger[acctID.ToMapKey()] = &staking.GenesisLedgerEntry{
-			GeneralBalance:  acct.GeneralBalance,
-			EscrowBalance:   acct.EscrowBalance,
-			DebondStartTime: acct.DebondStartTime,
-			Nonce:           acct.Nonce,
-		}
+		// All debonding stops on a dump/restore and needs to be restarted. This
+		// is also checked on chain init.
+		acct.Escrow.DebondingShares = *staking.NewQuantity()
```

Comment:
Ah I see.  I'm not sure people will be happy about this...

Reply:
Doesn't this eat tokens that are currently debonding?  The delegation's been updated, so the tokens held in debonding limbo need to either be put back, or this needs to be preserved no?

Reply:
Fixed so dump/restore now also dumps and restores unbonding delegations.